### PR TITLE
Ascent/SENSEI: Add Profiler

### PR DESCRIPTION
**Pull request by ax3l**
Add overall write profilers to Ascent & SENSEI.
Add detailed profile for sub-operations (blueprint, publish, execute) to Ascent as well.

### CODE REVIEW

File: Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp [hunk 1]
```diff
@@ -19,29 +19,48 @@ FlushFormatAscent::WriteToFile (
...
 #ifdef AMREX_USE_ASCENT
+    WARPX_PROFILE("FlushFormatAscent::WriteToFile()");
+    WARPX_PROFILE_VAR_NS("FlushFormatAscent::WriteToFile::MultiLevelToBlueprint",
+                         prof_ascent_mesh_blueprint);
+    WARPX_PROFILE_VAR_NS("FlushFormatAscent::WriteToFile::WriteParticles",
+                         prof_ascent_particles);
+    WARPX_PROFILE_VAR_NS("FlushFormatAscent::WriteToFile::publish",
+                         prof_ascent_publish);
+    WARPX_PROFILE_VAR_NS("FlushFormatAscent::WriteToFile::execute",
+                         prof_ascent_execute);
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp]: NS means "no start", not "namespace". So we can remove this: thx @kngott 
```suggestion
```

File: Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp [hunk 2]
```diff
@@ -19,29 +19,48 @@ FlushFormatAscent::WriteToFile (
...
+
     auto & warpx = WarpX::GetInstance();
 
     // wrap mesh data
+    WARPX_PROFILE_VAR_START(prof_ascent_mesh_blueprint);
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp]: If we remove the above, we write here:
```suggestion
    WARPX_PROFILE_VAR("FlushFormatAscent::WriteToFile::MultiLevelToBlueprint", prof_ascent_mesh_blueprint);
```

File: Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp [hunk 3]
```diff
@@ -19,29 +19,48 @@ FlushFormatAscent::WriteToFile (
...
     conduit::Node bp_mesh;
     amrex::MultiLevelToBlueprint(
         nlev, amrex::GetVecOfConstPtrs(mf), varnames, geom, time, iteration, warpx.refRatio(), bp_mesh);
+    WARPX_PROFILE_VAR_STOP(prof_ascent_mesh_blueprint);
 
+    WARPX_PROFILE_VAR_START(prof_ascent_particles);
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp]:
```suggestion
    WARPX_PROFILE_VAR("FlushFormatAscent::WriteToFile::WriteParticles", prof_ascent_particles);
```

File: Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp [hunk 4]
```diff
@@ -19,29 +19,48 @@ FlushFormatAscent::WriteToFile (
...
     WriteParticles(particle_diags, bp_mesh);
+    WARPX_PROFILE_VAR_STOP(prof_ascent_particles);
 
     // If you want to save blueprint HDF5 files w/o using an Ascent
     // extract, you can call the following AMReX helper:
     // const auto step = istep[0];
     // WriteBlueprintFiles(bp_mesh,"bp_export",step,"hdf5");
 
+    WARPX_PROFILE_VAR_START(prof_ascent_publish);
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp]:
```suggestion
    WARPX_PROFILE_VAR("FlushFormatAscent::WriteToFile::publish", prof_ascent_publish);
```

File: Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp [hunk 5]
```diff
@@ -19,29 +19,48 @@ FlushFormatAscent::WriteToFile (
...
     ascent::Ascent ascent;
     conduit::Node opts;
     opts["exceptions"] = "catch";
     opts["mpi_comm"] = MPI_Comm_c2f(ParallelDescriptor::Communicator());
     ascent.open(opts);
     ascent.publish(bp_mesh);
+    WARPX_PROFILE_VAR_STOP(prof_ascent_publish);
+
+    WARPX_PROFILE_VAR_START(prof_ascent_execute);
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp]:
```suggestion
    WARPX_PROFILE_VAR("FlushFormatAscent::WriteToFile::execute", prof_ascent_execute);
```

**Review by kngott — APPROVED** [review]

**Review by RemiLehe — APPROVED** [review]

File: Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp [hunk 6]
```diff
@@ -19,29 +19,40 @@ FlushFormatAscent::WriteToFile (
     bool /*isBTD*/, int /*snapshotID*/, const amrex::Geometry& /*full_BTD_snapshot*/, bool /*isLastBTDFlush*/) const
 {
 #ifdef AMREX_USE_ASCENT
+    WARPX_PROFILE("FlushFormatAscent::WriteToFile()");
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp:22]: Maybe needs same include as #2237 (`#include "Utils/WarpXProfilerWrapper.H"`)

Comment (ax3l) [reply Source/Diagnostics/FlushFormats/FlushFormatAscent.cpp:22]: Fix in #2238

**Pull request merged**